src/fsharp/ConstraintSolver.fs

```diff
@@ -2536,27 +2562,26 @@ and ResolveOverloading
                 match bestMethods with 
                 | [(calledMeth, warns, t)] -> Some calledMeth, OkResult (warns, ()), WithTrace t
                 | bestMethods -> 
-                    let methodNames =
-                        let methods = 
-                            // use the most precise set
-                            // - if after filtering bestMethods still contains something - use it
-                            // - otherwise use applicableMeths or initial set of candidate methods
-                            match bestMethods with
-                            | [] -> 
-                                match applicableMeths with
-                                | [] -> candidates
-                                | m -> m |> List.map (fun (x, _, _) -> x)
-                            | m -> m |> List.map (fun (x, _, _) -> x)
-
-                        methods
-                        |> List.map (fun cmeth -> NicePrint.stringOfMethInfo amap m denv cmeth.Method)
-                        |> List.sort
-                    let msg = FSComp.SR.csMethodIsOverloaded methodName
-                    let msg = 
-                        match methodNames with
-                        | [] -> msg
-                        | names -> sprintf "%s %s" msg (FSComp.SR.csCandidates (String.concat ", " names))
-                    None, ErrorD (failOverloading msg []), NoTrace
+                    let methods = 
+                        let getMethodSlotsAndErrors =
+                          function | methodSlot, []      -> List.singleton {methodSlot = methodSlot; error = Unchecked.defaultof<exn>; amap = amap}
```

Comment:
Nit - Anything involving `List.map` and `List.singleton` is almost always better as a list expression, e.g. this looks equivalent to:
```
let getMethodSlotsAndErrors (methodSlot, errors) =
    [ match errors with 
      | [] -> yield {methodSlot = methodSlot; error = Unchecked.defaultof<exn>; amap = amap}
      | errors  -> for error in errors do yield {methodSlot = methodSlot; error = error; amap = amap} ]
```